### ISSUE TITLE
.psd documents are all corrupted when I save them! "invalid file format u/000/000..."

### ISSUE THREAD

**Issue by norahsage** [photopea/photopea#3748]
All of these projects that I've worked really hard on are unreadable by any program, not just photopea. they ALL say they're 0 bytes and I cant get them to open to recover ANY of my work, I'm so upset. Is there any way to recover these documents?

I'm on windows 10, using the latest opera. Incognito window, Google chrome, and google chrome incognito photopea did not work either. Libre office draw and adobe also would not open them.

**Comment by photopea** [comment]
I think your files are lost, because files are empty (0 Bytes) :( There is a bug in Opera, which they refuse to fix.

Please, use a better browser, like Firefox or Safari, or Microsoft Edge.

**Comment by norahsage** [comment]
great to know now, after days worth of work is completely lost and I dont have time to redo it. awesome. 
theres no way to recover these files?!

**Comment by photopea** [comment]
Are you using Opera 79.0.4143.50? I just tried it and it seems like this bug is not present anymore (saved files never have zero bytes).

**Comment by photopea** [comment]
I am very sorry for it. Browsers are an important part of the whole experience, and when some browser has a bug, there is nothing we can do about it :(

**Comment by norahsage** [comment]
yep, its still doing it, which means the thing I'm currently working on is going to be lost, too. god dammit.

**Comment by norahsage** [comment]
currently attempting to copy over the layers I'm working on to a new project in chrome, hopefully that works.

**Comment by photopea** [comment]
Use File - Save PSD/PSB.

**Comment by norahsage** [comment]
unfortunately thats what corrupts the file, when I try to save it as a psd. Exporting as a jpg still works, but I need the working layers. 
transferring one by one to Chrome nearly seppuku'd my garbage cpu, but its worked for now. 

I imagine this has something to do with operas adblock feature?

**Comment by photopea** [comment]
The bug is with the saving API. Use File - Save PSD/PSB.

**Comment by norahsage** [comment]
nope, it just happened with Chrome as well. great.

**Comment by photopea** [comment]
What exactly is happening with Chrome?

**Comment by casperzwaart100** [comment]
> What exactly is happening with Chrome?

I can help with this one,

i used to use my chromebook for making some edits, sometimes it was really slow (obviously, it's a chromebook)
When it was being slow, and I saved a .psd file it'd be an empty file.

Now I recently switched to Mac, and am using it now. It's still being slow sometimes, not entirely sure why yet, but when I want to save a .psd now it just says "not enough RAM". So it's probably got something to do with not having enough memory to save the psd
